#3 - PDFExtract Output as String

wpmp opened the pull request:
- add processFileToString method that does not gzip output and returns it as string

wpmp committed:
#3 - PDFExtract Output as String

- add processFileToString method that does not gzip output and returns it as string

reckart commented on src/main/java/paperai/pdfextract/PDFExtractor.java:
```diff
@@ -44,6 +44,22 @@ public static void main(String[] args) {
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        BufferedOutputStream buf = new BufferedOutputStream(bos);
+        Writer w = new BufferedWriter(new OutputStreamWriter(bos, "UTF-8"));
```
Use a try-with-resource here:

```
try (Writer w = ...) {
 ...
}
```

That closes the write automatically when the block is left.

I believe you can also use `StandardCharsets.UTF_8` here in stead of a string literal.

The buffered stream is not necessary since you are writing to memory anyway. Buffering won't speed up anything.

wpmp replied:
Well I also didn't even reference the buffer I see now...

I will change it.

reckart replied:
Btw, I believe there is a `StringWriter` which may make this even simpler :)

wpmp replied:
I was too quick, already commited, should I still look into `StringWriter`?

reckart replied:
Your choice :) I'll approve it. You can still make an update to use the StringWriter before merging. Another approval will not be required.

wpmp replied:
I changed it to Stringwriter

wpmp committed:
#3 - PDFExtract Output as String

- use try-with-resource on writer
- remove buffer streams
- replace utf-8 string with constant

wpmp committed:
#3 PDFExtract Output as String

- replace writer by stringwriter and remove streams
